### PR TITLE
ss-13: Public and Private routing

### PR DESCRIPTION
**Pull request by teslmik**
6StXVcn7D8

### CODE REVIEW

**Comment by teslmik** [comment]
added route '/dashboard' for redirect with logged in user and state 'isLoggedIn' in auth slice

File: frontend/src/bundles/common/components/privat-route/privat-route.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Navigate } from 'react-router-dom';
+
+import { AppRoute } from '../../enums/enums';
+import { useAppSelector } from '../../hooks/hooks';
+
+const PrivatRoute: React.FC<{ component: React.FC }> = ({
+    component: Component,
+}) => {
+    const { isLoggedIn } = useAppSelector((state) => state.auth);
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/privat-route/privat-route.tsx:9]: It is better to check if we have fetched user

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/privat-route/privat-route.tsx (hunk 1 above)
Comment (teslmik) [reply frontend/src/bundles/common/components/privat-route/privat-route.tsx:9]: > It is better to check if we have fetched user

How can i do this? Login functionality not yet available

Comment (teslmik) [reply frontend/src/bundles/common/components/privat-route/privat-route.tsx]: - Fixed some logic
- Added a user to the authorization state
- Added user check
- Added a dashboard page and implemented routes for testing
- When registering, authorization occurs

**Review by nikita-remeslov — APPROVED** [review]

File: frontend/src/bundles/common/components/public-route/public-route.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    children,
+}) => {
+    const dispatch = useAppDispatch();
+    const { users } = useAppSelector(({ users }) => ({
+        users: users.users,
+    }));
+    const { user } = useAppSelector(({ auth }) => ({
+        user: auth.user,
+    }));
+    const isAuth = users.some((user_) => user_.id === user?.id);
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/public-route/public-route.tsx:18]: Not sure about this logic. You just need check user

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**